File: macos/Classes/VkNativeClientPlugin.swift
import Cocoa
import FlutterMacOS

// MIME types
let mimeTextPlain = "text/plain"
let mimeTextHtml = "text/html"

public class VkNativeClientPlugin: NSObject, FlutterPlugin {
    // Register the plugin with the Flutter engine
    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(name: "vk_native_client", binaryMessenger: registrar.messenger)
        let instance = VkNativeClientPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }
    
    // Handle method calls from Flutter
    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getClipboardData":
            result(ClipboardMacOS.getClipboardData()) // Get clipboard data
        case "setClipboardData":
            result(ClipboardMacOS.setClipboardData(call: call)) // Set clipboard data
        case "getClipboardDataMimeTypes":
            result(ClipboardMacOS.getClipboardDataMimeTypes()) // Get clipboard data MIME types
        default:
            result(FlutterMethodNotImplemented) // Method not implemented
        }
    }
    
    
}

File: android/src/main/kotlin/com/codevedas/vk_native_client/VkNativeClientPlugin.kt
// MIT License
//
// Copyright (c) 2023 Code Vedas
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal
// in the Software without restriction, including without limitation the rights
// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
// copies of the Software, and to permit persons to whom the Software is
// furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all
// copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
// SOFTWARE.
package com.codevedas.vk_native_client


import android.content.Context

import androidx.annotation.NonNull
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result


/** VkNativeClientPlugin */
class VkNativeClientPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel
    private lateinit var context: Context

    override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPluginBinding) {
        // Create a MethodChannel to communicate with Flutter
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "vk_native_client")
        channel.setMethodCallHandler(this)

        // Store the application context
        context = flutterPluginBinding.applicationContext
    }

    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
        // Handle method calls from Flutter
        when (call.method) {
            /// getClipboardData returns Map<String, String>
            "getClipboardData" -> result.success(ClipboardAndroid.getClipboardData(context))
            /// setClipboardData returns bool
            "setClipboardData" -> result.success(ClipboardAndroid.setClipboardData(call, context))
            /// getClipboardDataMimeTypes returns List<String>
            "getClipboardDataMimeTypes" -> result.success(ClipboardAndroid.getClipboardDataMimeTypes(context))
            else -> result.notImplemented() // Method not implemented
        }
    }

    override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
        // Clean up when detached from the Flutter engine
        channel.setMethodCallHandler(null)
    }
}